docs/source/design.rst

```diff
@@ -0,0 +1,109 @@
+        pseudo = UpfData.get_or_create(stream)
+
+If a new node was created, it will be unstored.
+Note that if more than one pseudopotential in the database is matched, a random one is selected and returned.
+
+Families
+========
+
+Having loose pseudopotentials floating around is not very practical, so groups of related pseudopotentials can be organized into "families", which are implemented as group plugins with the base class ``PseudoPotentialFamily``.
+A family class can in principle support many pseudopotential formats, however, a family instance can only contain pseudopotentials of a single format.
+For example, the ``PseudoPotentialFamily`` class supports all of the pseudopotential formats that are supported by this plugin.
+However, any instance can only contain pseudopotentials of the same format (e.g. *all* UPF or *all* PSP8, not a mixture).
+In contrast, the ``SsspFamily`` only supports the ``UpfData`` format.
+
+A pseudopotential family can be constructed manually, by first constructing the class instance and then adding pseudopotential data nodes to it:
+
+.. code-block:: python
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+    PseudoPotentialFamily = plugins.GroupFactory('pseudo')
+
+    pseudos = []
+
+    for filepath in ['Ga.upf', 'As.upf']:
+        with open(filepath, 'rb') as stream:
+            pseudo = UpfData(stream)
+            pseudos.append(pseudo)
+
+    family = PseudoPotentialFamily(label='pseudos/upf').store()
+    family.append(pseudos)
+
+Note that as with any ``Group``, it has to be stored before nodes can be added.
+If you have a folder on disk that contains various pseudopotentials for different elements, there is an even easier way to create the family automatically:
+
+.. code-block:: python
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+    PseudoPotentialFamily = plugins.GroupFactory('pseudo')
+
+    family = PseudoPotentialFamily('path/to/pseudos', 'pseudos/upf', pseudo_type=UpfData)
+
+The plugin is not able to reliably deduce the format of the pseudopotentials contained in the folder, so one should indicate what data type to use with the ``pseudo_type`` argument.
+The exception is when the family class only supports a single pseudo type, such as for the ``SsspFamily``, in which case that type will automatically be selected.
+Subclasses of supported pseudo types are also accepted.
+For example, the base class ``PseudoPotentialFamily`` supports pseudopotentials of the ``PseudoPotentialData`` type.
+Because all more specific pseudopotential types are subclasses of ``PseudoPotentialData``, the ``PseudoPotentialFamily`` class accepts all of them.
+
+Recommended cutoffs
+===================
+
+Certain pseudopotential family types, such as the ``SsspFamily``, provide recommended cutoff values for wave functions and charge density in plane-wave codes.
+These cutoffs are always in units of electronvolt.
```

Comment:
This is no longer true, unit can be set. Should probably have a link to the `pint` library to show which values for units are valid

Reply:
Right! I was going to open issues for adding more documentation for features such as the units, but I should indeed at least already correct this text.